From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 6 Jul 2016 09:06:37 -0400
Subject: dm rq: check kthread_run return for .request_fn request-based DM
References: bnc#1023798,FATE#321463
Patch-Mainline: v4.8-rc1
Git-commit: 7193a9defcab6f3d3f1eb64c68bad7534e5a39ad

Check return value of kthread_run() in dm_old_init_request_queue().

Reported-by: Minfei Huang <mnghuan@gmail.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/dm-rq.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/md/dm-rq.c b/drivers/md/dm-rq.c
index 266f7b6..aa81539 100644
--- a/drivers/md/dm-rq.c
+++ b/drivers/md/dm-rq.c
@@ -819,6 +819,8 @@ int dm_old_init_request_queue(struct mapped_device *md)
 	init_kthread_worker(&md->kworker);
 	md->kworker_task = kthread_run(kthread_worker_fn, &md->kworker,
 				       "kdmwork-%s", dm_device_name(md));
+	if (IS_ERR(md->kworker_task))
+		return PTR_ERR(md->kworker_task);
 
 	elv_register_queue(md->queue);
 
-- 
1.8.5.6

